iModelCore/ECPresentation/Tests/NonPublished/Integration/Hierarchies/HierarchyUpdateTests.cpp

```diff
@@ -10,4906 +10,1413 @@ USING_NAMESPACE_BENTLEY_SQLITE_EC
-    childNodes = RulesEngineTestHelpers::GetValidatedNodes([&](){ return m_manager->GetNodes(AsyncHierarchyRequestParams::Create(m_db, rules->GetRuleSetId(), RulesetVariables(), rootNodes[0].get())).get(); });
-    ASSERT_EQ(1, childNodes.GetSize());
-
-    // verify records
-    ASSERT_EQ(2, m_updateRecordsHandler->GetRecords().size());
-
-    EXPECT_TRUE(m_updateRecordsHandler->GetRecords()[0].GetParentNode().IsNull());
-    EXPECT_EQ(1, m_updateRecordsHandler->GetRecords()[0].GetNodesCount());
-    ASSERT_EQ(1, m_updateRecordsHandler->GetRecords()[0].GetExpandedNodes().size());
-    EXPECT_TRUE(m_updateRecordsHandler->GetRecords()[0].GetExpandedNodes()[0].GetNode()->Equals(*rootNodes[0]));
-
-    EXPECT_TRUE(m_updateRecordsHandler->GetRecords()[1].GetParentNode()->Equals(*rootNodes[0]));
-    EXPECT_EQ(1, m_updateRecordsHandler->GetRecords()[1].GetNodesCount());
-    ASSERT_EQ(0, m_updateRecordsHandler->GetRecords()[1].GetExpandedNodes().size());
-    }
-
-#ifdef wip_skipped_instance_keys_performance_issue
-/*---------------------------------------------------------------------------------**//**
-* @betest
-+---------------+---------------+---------------+---------------+---------------+------*/
-TEST_F (HierarchyUpdateTests, UpdateAfterSkippedOneToManyRelationshipDelete)
-    {
-    ECRelationshipClassCP relationshipClassWG = m_schema->GetClassCP("WidgetHasGadgets")->GetRelationshipClassCP();
-    ECRelationshipClassCP relationshipClassGS = m_schema->GetClassCP("GadgetHasSprockets")->GetRelationshipClassCP();
-
-    // insert a widget and a gadget
-    IECInstancePtr widget = RulesEngineTestHelpers::InsertInstance(m_db, *m_widgetClass);
-    IECInstancePtr gadget = RulesEngineTestHelpers::InsertInstance(m_db, *m_gadgetClass);
-    IECInstancePtr sprocket = RulesEngineTestHelpers::InsertInstance(m_db, *m_sprocketClass);
-
-    // relate the instances
-    ECInstanceKey relWGKey = RulesEngineTestHelpers::InsertRelationship(m_db, *relationshipClassWG, *widget, *gadget);
-    RulesEngineTestHelpers::InsertRelationship(m_db, *relationshipClassGS, *gadget, *sprocket, nullptr, true);
-
-    // create the rule set
-    PresentationRuleSetPtr rules = PresentationRuleSet::CreateInstance(BeTest::GetNameOfCurrentTest());
-    m_locater->AddRuleSet(*rules);
-
-    RootNodeRule* rule = new RootNodeRule("", 1, false, RuleTargetTree::TargetTree_Both, false);
-    rule->AddSpecification(*new InstanceNodesOfSpecificClassesSpecification(1, true, false, false, false, false, false,
-        "", "RulesEngineTest:Widget", false));
-    rules->AddPresentationRule(*rule);
-
-    ChildNodeRule* childRule = new ChildNodeRule();
-    childRule->AddSpecification(*new RelatedInstanceNodesSpecification(1, false, false, false, false, false, false, false, 1, "",
-        RequiredRelationDirection_Forward, "", "RulesEngineTest:GadgetHasSprockets", "RulesEngineTest:Sprocket"));
-    rules->AddPresentationRule(*childRule);
-
-    // request f
```

Comment:
let's also have a test for inserting/deleting the invisible `B` node

Reply:
Added tests to check inserting/removing/ nodes in hidden hierarchy level.